datadog/dogstatsd/base.py

```diff
@@ -274,24 +297,66 @@ def __init__(
         self._reset_telemetry()
         self._telemetry_flush_interval = telemetry_min_flush_interval
         self._telemetry = not disable_telemetry
+        self._last_flush_time = time.time()
 
+        self._current_buffer_total_size = 0
+        self._buffer = []  # type: List[Text]
         self._buffer_lock = RLock()
 
+        self._reset_buffer()
+
+        # This lock is used for backwards compatibility to prevent concurrent
+        # changes to the buffer when the user is managing the buffer themselves
+        # via deprecated `open_buffer()` and `close_buffer()` functions.
+        self._manual_buffer_lock = RLock()
+
+        # If buffering is disabled, we bypass the buffer function.
+        self._send = self._send_to_buffer
+        if disable_buffering is True:
+            log.info("Statsd buffering is disabled")
+            self._send = self._send_to_server
+
+        # Start the flush thread if buffering is enabled and the interval is above
+        # a reasonable range. This both prevents thrashing and allow us to use "0.0"
+        # as a value for disabling the automatic flush timer as well.
+        if disable_buffering is False and flush_interval >= MIN_FLUSH_INTERVAL:
+            self._register_flush_thread(flush_interval)
+            log.debug(
+                "Statsd flush thread registered with period of %s",
+                flush_interval,
+            )
+        else:
+            log.info("Statsd periodic buffer flush is disabled")
+
     def disable_telemetry(self):
         self._telemetry = False
 
     def enable_telemetry(self):
         self._telemetry = True
 
-    def _dedicated_telemetry_destination(self):
+    def _register_flush_thread(self, sleep_duration):
+        def _flush_thread_loop(self, sleep_duration):
+            while True:
+                time.sleep(sleep_duration)
+                self.flush()
+
+        self._flush_thread = threading.Thread(
+            name="{}_flush_thread".format(self.__class__.__name__),
+            target=_flush_thread_loop,
+            args=(self, sleep_duration,),
+        )
+        self._flush_thread.daemon = True
+        self._flush_thread.start()
+
+    def _dedicated_telemetry_dest(self):
```

Comment:
I am not sure why do we want to rename the method.
```suggestion
    def _dedicated_telemetry_destination(self):
```

Reply:
It was a change to fix a [Pylint warning C0103](http://pylint-messages.wikidot.com/messages:c0103). It effectively was saying that the name was invalid because its length was >=30 characters long and trimming it a bit seemed like an easy fix. If you want the old name back, that's fine - I was just trying to remove the pylint noise during development.

Reply:
I think we are fine with the old name for private method.

Reply:
Reverted to old name